src/source/lexbase.d

```diff
@@ -581,6 +600,17 @@ unittest {
 	checkLexInvalid("/*", "Comment must end with `*/`.");
 	checkLexInvalid("/*/", "Comment must end with `*/`.");
 
+	// EOF in the middle of the token
+	checkIncompleteInput("/*/", TokenType.Comment);
+	checkIncompleteInput("\"", TokenType.StringLiteral);
+	checkIncompleteInput("'", TokenType.CharacterLiteral);
+	checkIncompleteInput("0b", TokenType.IntegerLiteral);
+	checkIncompleteInput("0x", TokenType.IntegerLiteral);
+	checkIncompleteInput("0o", TokenType.IntegerLiteral);
+	checkIncompleteInput("1e", TokenType.FloatLiteral);
+	// EOF instead of a token
+	//checkIncompleteInput(" 	", TokenType.Comment);
```

Comment:
This test looks like it is completely decoupled from the rest, so why interleave it like this?

Reply:
not sure what you mean. If this is about it being in `lexbase`, then I moved all of the tests to `parserutil`